docs/backend/deprecation.md

```diff
@@ -0,0 +1,311 @@
+foo = None
+foo = deprecated(foo, 'foo is no more, use bar instead')
+```
+
+### Moving functions and classes
+
+Given we have a Python file at `old/foo/bar.py` and want to move some classes or functions to `new/baz/baaz.py`.
+
+Here `zope.deferredimport` offers a deprecation helper.
+It also avoids circular imports on initialization time.
+
+```python
+import zope.deferredimport
+zope.deferredimport.initialize()
+
+zope.deferredimport.deprecated(
+    "Import from new.baz.baaz instead",
+    SomeOldClass='new.baz:baaz.SomeMovedClass',
+    some_old_function='new.baz:baaz.some_moved_function',
+)
+
+def some_function_which_is_not_touched_at_all():
+    pass
+```
+
+### Deprecating a GenericSetup profile
+
+Starting with GenericSetup 1.8.2 (part of Plone > 5.0.2) the `post_handler` attribute in ZCML can be used to call a function after the profile was applied.
+We use this feature to issue a warning.
+
+First we register the same profile twice. Under the new name and under the old name:
+
+```xml
+<genericsetup:registerProfile
+    name="default"
+    title="My Fance Package"
+    directory="profiles/default"
+    description="..."
+    provides="Products.GenericSetup.interfaces.EXTENSION"
+    />
+
+<genericsetup:registerProfile
+    name="some_confusing_name"
+    title="My Fance Package (deprecated)"
+    directory="profiles/some_confusing_name"
+    description="... (use profile default instaed)"
+    provides="Products.GenericSetup.interfaces.EXTENSION"
+    post_handler=".setuphandlers.deprecate_profile_some_confusing_name"
+    />
+```
+
+And in `setuphandlers.py` add a function:
+
+```python
+import warnings
+
+def deprecate_profile_some_confusing_name(tool):
+    warnings.warn(
+        'The profile with id "some_confusing_name" was renamed to "default".',
+        DeprecationWarning
+    )
+```
+
+### Deprecating a template position
+
+Sometimes we need to move templates to new locations. Since addons often use jbot to override templates by their position, we need to point them to the new position as well as make sure that the override still works with the old position.
+
+
+In the old package folders `__init__.py` we add a mapping dict with the name `jbot_deprecations` of old and new template location:
+
+```python
+jbot_deprecations = {
+    "plone.locking.browser.info.pt": "plone.app.layout.viewlets.info.pt"
+}
+```
+
+add a jbot-deprecations config to the `configure.zcml` file
+
+create a `jbot_deprecations.zcml` file:
+
+```{code-block} xml
+:emphasize-lines: 6,9-12
+:linenos:
+
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    xmlns:zcml="http://namespaces.zope.org/zcml"
+    >
+
+  <configure zcml:condition="have jbot-deprecations">
+    <include package="z3c.jbot" file="meta.zcml" />
+    <browser:jbotDeprecated dictionary=".jbot_deprecations" />
+  </configure>
```

Comment:
```suggestion
  <include
      zcml:condition="installed z3c.jbot"
      package="z3c.jbot" 
      file="meta.zcml" 
      />
  <browser:jbotDeprecated
      zcml:condition="have jbot-deprecations"
      dictionary=".jbot_deprecations"
      />
```

Reply:
I think that the `meta.zcml` has to be loaded before checking the feature. If we have the `jbot-deprecations` feature, it means the meta has been already loaded.

CC @mauritsvanrees